sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4137,15 +4137,14 @@ def post_teardown_cleanup(self,
         if terminate:
             cloud = handle.launched_resources.cloud
             config = common_utils.read_yaml(handle.cluster_yaml)
-            # TODO(tian): Add some API like
-            # provision_lib.supports(cloud, 'cleanup_ports')
-            # so that our backend do not need to know the specific details
-            # for different clouds.
-            if (cloud.PROVISIONER_VERSION >= clouds.ProvisionerVersion.
-                    RAY_PROVISIONER_SKYPILOT_TERMINATOR or
```

Comment:
Looks like this was recently added, not sure if it's necessary along with `if provision_lib.supports(repr(cloud), 'cleanup_ports')`

Reply:
cc @cblmemo - can you take a quick look at this?

Reply:
This is added in #2681, https://github.com/skypilot-org/skypilot/pull/2681/files#diff-f14940b9d929546868fb41cb0b27a1db08b321b7bbc0cb3ab3b864589aa82aa8
I still think the `provision_lib.supports(cloud, 'cleanup_ports')` is needed since `clouds.ProvisionerVersion` is not related to what cloud supports open ports. The new condition here is essentially still `isinstance(cloud, (GCP, AWS, Azure))`. The current implementation LGTM; cc @Michaelvll for a double-check

Reply:
How about we change the check to the following:
```python
try:
    cloud.check_features_are_supported(handle.launched_resources, {clouds.CloudImplementationFeatures.OPEN_PORTS})
    provision_lib.cleanup_ports(repr(cloud),
                                                cluster_name_on_cloud,
                                                handle.launched_resources.ports,
                                                config['provider'])
except exceptions.NotSupportedError:
    pass
except ...
```

Reply:
Agreed! Let's use `cloud.check_features_are_supported`. This should be fine since all ports related functions is implemented in new provisioned API.